### PR TITLE
 Speeds up preferences menu, mob dummies, & Unit Tests outfits

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

This PR puts loading preferences menu stuff on a subsytem, SSwardrobe, which allows us to preload stuff like equipment clothing and organs of dummies, and load it when opening the menu, saving us a lot of time.

Additionally, we now unit test outfits

Ports:
- https://github.com/tgstation/tgstation/pull/44109
- https://github.com/tgstation/tgstation/pull/47504
- https://github.com/tgstation/tgstation/pull/52210
- https://github.com/tgstation/tgstation/pull/59109 (wardrobe changes only)
- https://github.com/tgstation/tgstation/pull/63225
- https://github.com/tgstation/tgstation/pull/85945

## Why It's Good For The Game

prequel to #11340 

Loading dummies in the preference menu does not make them take an ungodly amount of time just because they have more gear on. (Looking at you, Chief Engineer)

If you want measurement specifics, go look at the TG pr.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Basically instant

https://github.com/user-attachments/assets/85cdab9c-0117-48d3-943c-958bc43f7710

</details>

## Changelog
:cl: rkz, Akrilla, Mothblocks, LemonInTheDark, Firecage, MrMelbert
refactor: Preferences menu will now re-use shared wardrobe items, making previews update faster.
code: Adds unit test for the wardrobe system
refactor: Refactors survivalboxes to be less copypaste
code: Standardizes storage holdability
fix: Fixes runtime when loading dummies in prefs or select-equipment context verb
/:cl:

### CODE REVIEW

**Review by Dejaku51 — CHANGES_REQUESTED** [review]
Doublecheck if you didn't skip those paths
/obj/item/storage/belt/medical/ert/ComponentInitialize()
/obj/item/storage/belt/bandolier/western/ComponentInitialize()
/obj/item/storage/firstaid/medical/paramedic/ComponentInitialize()

**Comment by Tsar-Salat** [comment]
> /obj/item/storage/belt/medical/ert/ComponentInitialize()

It just adds to its parent's list. The setter should work fine.

I took the opportunity to run it through the preloader though.



> /obj/item/storage/belt/bandolier/western/ComponentInitialize()

Missed this one. Non-conventional naming. Fixed now.



> /obj/item/storage/firstaid/medical/paramedic/ComponentInitialize()

Screenshot 2024-11-13 005554

Escaped my notice. Fixed

File: code/modules/surgery/bodyparts/bodyparts.dm [hunk 1]
```diff
@@ -500,11 +500,12 @@
 		C = owner
 		no_update = FALSE
 
-	if(HAS_TRAIT(C, TRAIT_HUSK) && IS_ORGANIC_LIMB(src))
-		dmg_overlay_type = "" //no damage overlay shown when husked
-		is_husked = TRUE
-	else
-		is_husked = FALSE
+	if(IS_ORGANIC_LIMB(src))
```
Comment (Tsar-Salat) [inline code/modules/surgery/bodyparts/bodyparts.dm:503]: Included additional fix.

Runtime exited here due to undefined 'status_traits' (`HAS_TRAIT` check) when loading dummies through select equipment.

Fixed by nesting the checks by 1.

**Review by Dejaku51 — APPROVED** [review]
looks good

**Review by PowerfulBacon — APPROVED** [review]

**Comment by itsmeow** [comment]
hold up

**Review by itsmeow — APPROVED** [review]
ok
